presto-main/src/main/java/io/prestosql/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -504,10 +518,12 @@ public PlanNode visitJoin(JoinNode node, RewriteContext<Expression> context)
                     newJoinFilter.isPresent() == node.getFilter().isPresent() &&
                             (!newJoinFilter.isPresent() || areExpressionsEquivalent(newJoinFilter.get(), node.getFilter().get()));
 
+            PlanNode output = node;
             if (leftSource != node.getLeft() ||
                     rightSource != node.getRight() ||
                     !filtersEquivalent ||
-                    !ImmutableSet.copyOf(equiJoinClauses).equals(ImmutableSet.copyOf(node.getCriteria()))) {
+                    !dynamicFilters.equals(node.getDynamicFilters()) ||
```

Comment:
I think `!dynamicFilters.equals(node.getDynamicFilters())` check is redundant here to `!equiJoinClausesEquivalent` check.

Reply:
It's possible that equijoin clause didn't change in visitJoin of PredicatePushdown but it's different from the clause which was used to create current dynamic filters. Although we will create the dynamic filters again from the new clause, new join node would not be created unless we checked for change in dynamic filters.